RxSwift/Observables/Max.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  Max.swift
+//  RxSwift
+//
+//  Created by Shai Mishali on 8/19/17.
+//  Copyright © 2017 Krunoslav Zaher. All rights reserved.
+//
+
+extension ObservableType {
+
+    /**
+     Emits the element from the source Observable that had the maximum value using a provided comparator closure.
+
+     - parameter comparator: A comparator method expected to return true if the first element has a greater value then the second element.
+     - returns: An observable sequence containing the specified number of elements from the end of the source sequence.
+    */
+    public func max(_ comparator: @escaping (E, E) -> Bool)
```

Comment:
In general, you're right. And I also sort of Contemplated if I could just have sort of a MinMaxSink with discrete implementations in the same file, but : 

1. I didn't know if that is the consensus in the RxSwift repo. It sort of seems most operators get their own file. 
2. Looking at Debounce and Throttle, they are very similar but they still get a file each. I guess the point is you could change implementations independently etc but TBH I really don't mind consolidating those if @kzaher will feel that's needed